docs/security/authentication/index.asciidoc

```diff
@@ -8,11 +8,34 @@
 {kib} supports the following authentication mechanisms:
 
 - Basic Authentication
+- Token Authentication
 - SAML Single Sign-On
 
 ==== Basic Authentication
 
-Basic Authentication requires a username and password to successfully log in to {kib}. It is enabled by default and based on the Native security realm provided by {es}. For more information about Basic Authentication and built-in users, see {xpack-ref}/setting-up-authentication.html[Setting Up User Authentication].
+Basic authentication requires a username and password to successfully log in to {kib}. It is enabled by default and based on the Native security realm provided by {es}. The basic authentication provider uses a Kibana provided login form, and supports authentication using the `Authorization` request header's `Basic` scheme. The session cookies which are issued by the basic authentication provider are stateless, and as such logging out of Kibana when using the basic authentication provider does not invalidate the session cookie for reuse.
+
+For more information about basic authentication and built-in users, see {xpack-ref}/setting-up-authentication.html[Setting Up User Authentication].
+
+==== Token Authentication
+
+Token authentication allows users to login using the same Kibana provided login form as Basic authentication, and supports authentication using the `Authorization` request header's `Bearer` scheme. The token authentication provider is built upon {es}'s {ref}/security-api-get-token.html[token APIs]. The session cookies which are issued by the Token authentication provider are stateful, and logging out of Kibana invalidates the session cookies for reuse. See {ref}/security-api-get-token.html#_description_86[Elasticsearch's token API] documentation to ensure token support is enabled.
+
+To enable the token authentication provider in Kibana, set the following value in your `kibana.yml`:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+xpack.security.authProviders: [token]
+--------------------------------------------------------------------------------
+
+It's also possible to enable both the token authentication provider in addition to the basic authentication provider. Whichever provider is specified first will be used by the Kibana provided login form and the subsequent authentication provider will only be used when the `Authorization` request header's scheme matches the provider.
+
+To make the token authentication provider used by the login form, but also add support for the `Authorization` request header with the `Basic` scheme, set the following in your `kibana.yml`:
```

Comment:
__optional nit:__ what do you think about mentioning an example use case for this setup (e.g. to use with `curl`), like we did for `saml` at the very bottom of this page?

Reply:
That makes complete sense also, revisions coming!